src/chain/chaininterface.rs

```diff
@@ -45,16 +45,20 @@ pub trait ChainWatchInterface: Sync + Send {
 	/// Indicates that a listener needs to see all transactions.
 	fn watch_all_txn(&self);
 
-	/// Register the given listener to receive events. Only a weak pointer is provided and the
-	/// registration should be freed once that pointer expires.
-	fn register_listener(&self, listener: Weak<ChainListener>);
-	//TODO: unregister
-
 	/// Gets the script and value in satoshis for a given unspent transaction output given a
 	/// short_channel_id (aka unspent_tx_output_identier). For BTC/tBTC channels the top three
 	/// bytes are the block height, the next 3 the transaction index within the block, and the
 	/// final two the output within the transaction.
 	fn get_chain_utxo(&self, genesis_hash: Sha256dHash, unspent_tx_output_identifier: u64) -> Result<(Script, u64), ChainError>;
+
+	/// Gets the list of transactions and transaction indices that the ChainWatchInterface is
+	/// watching for.
+	fn get_watched_txs(&self, block: &Block) -> (Vec<Transaction>, Vec<u32>);
+
+	/// Provides a way for users of the ChainWatchInterface to determine whether its watched
```

Comment:
This doc doesn't say much of anything about the semantics of it, nor how to use it.

Reply:
lmk if it's improved